src/transformState.js

```diff
@@ -46,7 +46,33 @@
  * @param {Object[]} transforms
  */
 function transformState(state, transforms) {
-  // write code here
+  if (state.length === 0) {
+    return state;
+  }
+
+  for (const obj in transforms) {
+    const currentTransform = transforms[obj];
+
+    switch (currentTransform.operation) {
+      case 'addProperties':
+        Object.assign(state, currentTransform.properties);
+        break;
+
+      case 'removeProperties':
+        for (const key of currentTransform.properties) {
+          delete state[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+    }
```

Comment:
You can add default: break; to this switch for best practice, even though it is not required with current tests